lang/cs/Org.Apache.REEF.Network/Elastic/Failures/Default/DefaultFailureStateMachine.cs

```diff
@@ -0,0 +1,320 @@
+            if (!weights.All(weight => weight.Item1 is DefaultFailureState))
+            {
+                throw new ArgumentException("Input is not of type DefaultFailureStateMachine,");
+            }
+
+            if (weights.Any(weight => weight.Item1.FailureState == (int)DefaultFailureStates.Continue))
+            {
+                throw new ArgumentException("Cannot change the threshold for Continue state.");
+            }
+
+            lock (_statusLock)
+            {
+                foreach (Tuple<IFailureState, float> weight in weights)
+                {
+                    transitionWeights[(DefaultFailureStates)weight.Item1.FailureState] = weight.Item2;
+                }
+
+                CheckConsistency();
+            }
+        }
+
+        /// <summary>
+        /// Utility method used to clone the target failure machine.
+        /// Only the thresholds are cloned, while the machine state is not.
+        /// </summary>
+        /// <param name="initalPoints">How many data points are avaialble in the new state machine</param>
+        /// <param name="initalState">The state from which the new machine should start</param>
+        /// <returns>A new failure machine with the same settings</returns>
+        public IFailureStateMachine Clone(
+            int initalPoints = 0, 
+            int initalState = (int)DefaultFailureStates.Continue)
+        {
+            var newMachine = new DefaultFailureStateMachine(initalPoints, (DefaultFailureStates)initalState);
+
+            foreach (DefaultFailureStates state in transitionWeights.Keys.OrderByDescending(x => x))
+            {
+                newMachine.SetThreshold(new DefaultFailureState((int)state), transitionWeights[state]);
+            }
+
+            return newMachine;
+        }
+
+        /// <summary>
+        /// Check if the states and related thresholds and consistent: i.e., each state can move 
+        /// up or down to only one other state.
+        /// </summary>
+        private void CheckConsistency()
+        {
+            lock (_statusLock)
+            {
+                var state = DefaultFailureStates.ContinueAndReconfigure;
+                float prevWeight = transitionWeights[state];
+                state = transitionMapUp[state];
+                float nextWeight = transitionWeights[state];
+
+                while (nextWeight >= 0)
+                {
+                    if (nextWeight < prevWeight)
+                    {
+                        throw new IllegalStateException(
+                            $"State {transitionMapDown[state]} weight is bigger than state {state}.");
+                    }
+
+                    prevWeight = nextWeight;
+
+                    if (state == DefaultFailureStates.StopAndReschedule)
+                    {
+                        return;
+                    }
+
+                    state = transitionMapUp[state];
+                    transitionWeights.TryGetValue(state, out nextWeight);
```

Comment:
we can move that call to `while` condition, i.e.
```C#
// float nextWeight = transitionWeights[state]; <-- remove that line, we do it in while now
while (transitionWeights.TryGetValue(state, out float nextWeight))
```

Reply:
Yep. Fixed.